tools/clang/lib/Sema/SemaHLSLDiagnoseTU.cpp

```diff
@@ -295,6 +301,39 @@ clang::FunctionDecl *ValidateNoRecursion(CallGraphWithRecurseGuard &callGraph,
   return nullptr;
 }
 
+bool DiagnoseHLSLMethodCall(CXXMemberCallExpr *CE, clang::Sema *self) {
```

Comment:
I do not like how this function is named exactly the same as the function "DiagnoseHLSLMethodCall" in SemaHLSL.cpp. I would think we'd either want to make one function, or leave these two functions as they are except rename them so that we know how they differ.

Reply:
Fixed. Now there're only 1 DiagnoseHLSLMethodCall.